### ISSUE TITLE
Rule against redundant closures

### ISSUE THREAD

**Issue by not-my-profile** [denoland/deno_lint#1169]
E.g. invalid:

```ts
xs.map(x => foo(x))
```

Valid:

```ts
xs.map(foo)
```

**Comment by magurotuna** [comment]
This seems more difficult for me than it seems to be, because for example the following two examples behave _differently_:

```ts
['10', '10', '10'].map(parseInt);
// [ 10, NaN, 2 ]
```

```ts
['10', '10', '10'].map((x) => parseInt(x));
// [ 10, 10, 10 ]
```

I'd say 99.9% of the cases what people want to do is the second one, not first one. Also the linter should not suggest the change that affects the code behavior in runtime. So the linter would need to be smart enough to consider the second example to be _not redundant_ - this sounds really hard.

**Comment by not-my-profile** [comment]
Ah, very good call! ([explanation for others wondering what's happening](https://stackoverflow.com/a/262511)).

You're right, doing this reliably would require type awareness, which is currently blocked as explained in #1138 ... so I'm closing this for now.
